zetaclient/zetacore/client_monitor.go

```diff
@@ -112,15 +114,19 @@ func (c *Client) MonitorVoteInboundResult(
 	}()
 
 	call := func() error {
-		err := c.monitorVoteInboundResult(ctx, zetaTxHash, retryGasLimit, msg)
+		err := c.monitorVoteInboundResult(ctx, zetaTxHash, retryGasLimit, msg, monitorErrCh)
 
 		// force retry on err
 		return retry.Retry(err)
 	}
 
+	//                               10 attempts,    2 seconds,     4 seconds max
+	// This will retry for a maximum of ~40 seconds with exponential backoff,
+	// However, this call is recursive for up to 1 layer and so the maxim time this can take is ~80 seconds
 	err := retryWithBackoff(call, monitorRetryCount, monitorInterval, monitorInterval*2)
 	if err != nil {
-		logger.Error().Err(err).Msg("unable to query tx result")
+		// All errors are forced to be retryable, we only return an error if the tx result cannot be queried
+		logger.Error().Err(err).Str(logs.FieldZetaTx, zetaTxHash).Msg("unable to query tx result")
```

Comment:
The main reason for reaching this condition , is that the mempool is congested , 
In this situation we are still creating an internal tracker which would create more imbound votes and pile up the mempool even more ?

Reply:
The internal tracker is to keep the missed inbound in memory.

We can still customize the logic of iterating these trackers, like not iterating if mempool is full.